pkg/server/cluster_server.go

```diff
@@ -129,6 +140,18 @@ func (cs *clusterServer) GetConfig(cr poolRequest) (*runtime.RawExtension, error
 		return nil, fmt.Errorf("could not get controllerconfig: %w", err)
 	}
 
+	// we cannot mock this in a test env
+	if cs.configMapLister != nil {
+		cm, err := cs.configMapLister.ConfigMaps(ctrlcommon.MCONamespace).Get("kubeconfig-data")
+		if err != nil {
+			klog.Errorf("Could not get kubeconfig data: %v", err)
+		} else {
+			kc := cm.BinaryData["kubeconfig"]
+			addDataAndMaybeAppendToIgnition("/etc/kubernetes/kubeconfig", kc, &ignConf)
+			cs.kubeconfigFunc = nil
```

Comment:
This works but I'm wondering if for future purposes it might be easier not to have 2 places that could set this file, and just lift it all into this logic here?

Reply:
I am most likely going to do that in the non-backported version, yeah. For now I think this is clean enough as a workaround